### PR TITLE
Add _latex_ method for hyperelliptic curves

### LINKED ISSUE #38386: HyperellipticCurve_generic doesn't define _latex_
### Steps To Reproduce

Steps:
1. Define and try to print and display a hyperelliptic curve
```python
F.<x> = QQ[]
F = x^2 + x + 1

C = HyperellipticCurve(F)

print(C)
display(C)
```

### Expected Behavior

Both the print and display output the same content, with `display` having better formatting using LaTeX.

### Actual Behavior

`display` is falling back to the default LaTeX representation for generic curves because `_latex_` is not defined for hyperelliptic curves.

In the above example, `print(C)` gives
```
Hyperelliptic Curve over Rational Field defined by y^2 = x^2 + x + 1
```
which is a good way to print out the curve.

However `display(C)` gives (I'm showing the LaTeX code for the output since GitHub doesn't support LaTeX):

```
\text{Projective Plane curve over $\Bold{Q}$ defined by $-x_{0}^{2} + x_{1}^{2} - x_{0} x_{2} - x_{2}^{2}$}
```
which is not as readable as what you get from printing the curve.

### Additional Information

_No response_

### Environment

```markdown
- **OS**: Fedora 40
- **Sage Version**: 10.4rc2, running in a system Jupyter notebook installation
```

### Checklist

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.
- [X] I have read the documentation and troubleshoot guide

### PR DESCRIPTION
**Pull request by vincentmacri**
Fixes #38386.

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation and checked the documentation preview.

### :hourglass: Dependencies

None

### CODE REVIEW

**Comment by grhkm21** [comment]
CI errors seem irrelevant.

File: src/sage/schemes/hyperelliptic_curves/hyperelliptic_generic.py [hunk 1]
```diff
@@ -183,6 +183,34 @@ def _repr_(self):
...
+            \text{Hyperelliptic Curve over $\Bold{Q}$ defined by $v^{2} = 4 u^{5} - 30 u^{3} + 45 u - 22$}
+            sage: C = HyperellipticCurve(x^5 + 1, x^2 + 3); latex(C)
+            \text{Hyperelliptic Curve over $\Bold{Q}$ defined by $y^{2} + \left(x^{2} + 3\right) y = x^{5} + 1$}
+        """
+
+        f, h = self._hyperelliptic_polynomials
+        R = self.base_ring()
+        y = self._printing_ring.gen()
+        x = self._printing_ring.base_ring().gen()
+        if h == 0:
```
Comment (grhkm21) [inline src/sage/schemes/hyperelliptic_curves/hyperelliptic_generic.py]:
```suggestion
        if h.is_zero():
```

File: src/sage/schemes/hyperelliptic_curves/hyperelliptic_generic.py [hunk 2]
```diff
@@ -183,6 +183,34 @@ def _repr_(self):
...
+            return (fr'\text{{Hyperelliptic Curve over ${R._latex_()}$ '
+                    f'defined by ${(y**2)._latex_()} = {(f(x))._latex_()}$}}')
+        else:
+            return (fr'\text{{Hyperelliptic Curve over ${R._latex_()}$ '
+                    f'defined by ${(y**2)._latex_()} + {(h(x)*y)._latex_()} = '
+                    f'{(f(x))._latex_()}$}}')
```
Comment (grhkm21) [inline src/sage/schemes/hyperelliptic_curves/hyperelliptic_generic.py]:
```suggestion
        return (fr'\text{{Hyperelliptic Curve over ${R._latex_()}$ '
                f'defined by ${(y**2)._latex_()} + {(h(x)*y)._latex_()} = '
                f'{(f(x))._latex_()}$}}')
```

**Review by grhkm21 — APPROVED** [review]
small changes then it's ok

**Comment by vincentmacri** [comment]
Thanks for the good suggestions. I copied this code from the `_repr_` method. I'll make the same changes to the `_repr_` method while I'm at this.

**Pull request merged**